Signal/src/ViewControllers/MessageComposeTableViewController.m

```diff
@@ -654,12 +685,36 @@ - (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath
     } else {
         OWSAssert(indexPath.section == MessageComposeTableViewControllerSectionContacts)
 
-        NSString *identifier = [[[self contactForIndexPath:indexPath] textSecureIdentifiers] firstObject];
+            if (!self.searchController.active && self.hasNoContacts)
+        {
+            // TODO: We could do something here, like show the "invite contacts" UI.
+            return;
+        }
+
+        Contact *contact = [self contactForIndexPath:indexPath];
+        NSString *contactIdentifier = [[contact textSecureIdentifiers] firstObject];
+
+        __weak MessageComposeTableViewController *weakSelf = self;
+        void (^newContactConversationBlock)() = ^{
+            [weakSelf dismissViewControllerAnimated:YES
+                                         completion:^() {
+                                             [Environment messageIdentifier:contactIdentifier withCompose:YES];
+                                         }];
+        };
 
-        [self dismissViewControllerAnimated:YES
-                                 completion:^() {
-                                     [Environment messageIdentifier:identifier withCompose:YES];
-                                 }];
+        if ([self isContactBlocked:contact]) {
```

Comment:
It appears that you're asking the user to unblock the contact before you let them show the message view for that user. I don't think we should do this.

For one, I don't think its helpful. We *already* have a way of doing this by unblocking upon send.

Secondly, message compose contact picker is a way to "search" for old threads by contact. e.g. if I want to view my old thread with Bob, without scrolling around trying to find it in the inbox/archive, I come to the compose view and type "Bo..." and tap on Bob's avatar, which brings me to my existing thread with Bob. I shouldn't have to unblock him at this point before I can view our old thread.